File: src/main/java/pl/mroz/buddiesapi/domain/account/Account.java
package pl.mroz.buddiesapi.domain.account;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import pl.mroz.buddiesapi.domain.common.Location;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Account {

    private UUID accountId;
    private String email;
    private String hashedPassword;
    private String name;
    private String lastName;
    private Location location;
    // todo rethink domain scopes and db schema

    public static Account fromDb(AccountRepository.IAccountEntity entity) {
        return Account.builder()
                .accountId(entity.getAccountId())
                .email(entity.getEmail())
                .hashedPassword(entity.getHashedPassword())
                .name(entity.getName())
                .lastName(entity.getLastName())
                .location(Location.fromDb(entity.getLocationEntity()))
                .build();
    }
}

File: src/main/kotlin/buddiesapi/domain/account/Account.kt
package buddiesapi.domain.account

import buddiesapi.domain.common.Location
import java.util.UUID

data class Account(
    val accountId: UUID,
    val email: String,
    val hashedPassword: String,
    val name: String,
    val lastName: String,
    val location: Location
) {

    companion object {
        fun fromDb(entity: AccountRepository.IAccountEntity): Account {
            return entity.let {
                Account(
                    it.accountId!!,
                    it.email,
                    it.hashedPassword,
                    it.name,
                    it.lastName,
                    Location.fromDb(it.locationEntity)
                )
            }

        }
    }
}
